scripts/Game/Core/Enums & Containers/CSI_PlayerData.c

```diff
@@ -0,0 +1,262 @@
+	{
+		m_iColorTeam = colorTeam;
+	}
+
+	//------------------------------------------------------------------------------------------------
+	void SetOverrideIcon(CSI_EOverrideIcon overrideIcon)
+	{
+		m_iOverrideIcon = overrideIcon;
+	}
+
+	//------------------------------------------------------------------------------------------------
+	void SetDisplayIcon(CSI_EIcon displayIcon)
+	{
+		m_iDisplayIcon = displayIcon;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	void SetRank(SCR_ECharacterRank rank)
+	{
+		m_iRank = rank;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	void SetIsTeamLeader(bool isTL)
+	{
+		m_bIsTeamLeader = isTL;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	void SetIsSquadLeader(bool isSL)
+	{
+		m_bIsSquadLeader = isSL;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	// GETTERS
+	//------------------------------------------------------------------------------------------------
+	
+	//------------------------------------------------------------------------------------------------
+	int GetPlayerValue()
+	{
+		return m_iPlayerValue;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	CSI_EColorTeam GetColorTeam()
+	{
+		return m_iColorTeam;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	CSI_EOverrideIcon GetOverrideIcon()
+	{
+		return m_iOverrideIcon;
+	}
+
+	//------------------------------------------------------------------------------------------------
+	CSI_EIcon GetDisplayIcon()
+	{
+		return m_iDisplayIcon;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	SCR_ECharacterRank GetRank()
+	{
+		return m_iRank;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	bool GetIsTeamLeader()
+	{
+		return m_bIsTeamLeader;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	bool GetIsSquadLeader()
+	{
+		return m_bIsSquadLeader;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	ScriptInvoker GetOnDataUpdate()
+	{
+		if (!m_OnDataUpdate)
+			m_OnDataUpdate = new ScriptInvoker();
+
+		return m_OnDataUpdate;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	// REPLICATION STUFF
+	//------------------------------------------------------------------------------------------------
+	
+	//------------------------------------------------------------------------------------------------
+	bool RplSave(ScriptBitWriter writer)
+	{
+		writer.Write(m_iColorTeam, 6);
```

Comment:
Are you sure these bit sizes are correct? We use 6 **bits** for RplSave, but 4 **bytes** for Extract? Sounds like packet corruption that the engine _may_ try to handle.

I'd recommend defining constants at the top of the file and using those so this is streamlined.